ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/load/table/LoadTable.java

```diff
@@ -51,18 +55,28 @@ Licensed to the Apache Software Foundation (ASF) under one
   private final static Logger LOG = LoggerFactory.getLogger(LoadTable.class);
   //  private final Helper helper;
   private final Context context;
+  private final ReplLogger replLogger;
   private final TableContext tableContext;
   private final TaskTracker tracker;
   private final TableEvent event;
 
-  public LoadTable(TableEvent event, Context context, TableContext tableContext, TaskTracker limiter)
+  public LoadTable(TableEvent event, Context context, ReplLogger replLogger,
+                   TableContext tableContext, TaskTracker limiter)
       throws SemanticException, IOException {
     this.event = event;
     this.context = context;
+    this.replLogger = replLogger;
     this.tableContext = tableContext;
     this.tracker = new TaskTracker(limiter);
   }
 
+  private void createTableReplLogTask(String tableName, TableType tableType) throws SemanticException {
+    ReplStateLogWork replLogWork = new ReplStateLogWork(replLogger,tableName, tableType);
+    Task<ReplStateLogWork> replLogTask = TaskFactory.get(replLogWork, context.hiveConf);
+    ReplLoadTask.dependency(tracker.tasks(), replLogTask);
+    tracker.addTask(replLogTask);
```

Comment:
I dont think we can directly add the replLogTask to tracker as it will be added as root task. Essentially this should be added as a dependency task to the DAG which is done earlier statement, but rather than taking the DAG from tracker looks like this has to be maintained separately so the dependency can be added and then the DAG root added to tracker